src/authentication/msal-browser/MSALBrowserAuthenticationProvider.ts

```
@@ -0,0 +1,73 @@
+	/**
+	 * @public
+	 * @constructor
+	 * Creates an instance of ImplicitMSALAuthenticationProvider
+	 * @param {PublicClientApplication} msalApplication - An instance of MSAL PublicClientApplication
+	 * @param {MSALBrowserAuthenticationProviderOptions} options - An instance of MSALAuthenticationProviderOptions
+	 * @returns An instance of ImplicitMSALAuthenticationProvider
+	 */
+	public constructor(private publicClientApplication: PublicClientApplication, private options: MSALBrowserAuthenticationProviderOptions) {
+		if (!this.options || !this.publicClientApplication) {
+			throw new GraphClientError("Please pass valid PublicClientApplication instance and MSALBrowserAuthenticationProviderOptions instance to instantiate MSALBrowserAuthenticationProvider");
+		}
+		this.options = options;
+		this.publicClientApplication = publicClientApplication;
+	}
+
+	/**
+	 * @public
+	 * @async
+	 * To get the access token for the request
+	 * @returns The promise that resolves to an access token
+	 */
+	public async getAccessToken(): Promise<string> {
+		const scopes = this.options && this.options.scopes;
+		const account = this.options && this.options.account;
+		const error = new GraphClientError();
+		if (!scopes || scopes.length === 0) {
+			error.name = "Empty Scopes";
+			error.message = "Scopes cannot be empty, Please provide scopes";
+			throw error;
+		}
+		try {
+			const response: AuthenticationResult = await this.publicClientApplication.acquireTokenSilent({
+				scopes,
+				account,
+			});
+			return response.accessToken;
+		} catch (error) {
+			if (error instanceof InteractionRequiredAuthError) {
+				if (this.options.interactionType === InteractionType.Redirect) {
+					this.publicClientApplication.acquireTokenRedirect({ scopes });
+				} else if (this.options.interactionType === InteractionType.Popup) {
+					const response: AuthenticationResult = await this.publicClientApplication.acquireTokenPopup({ scopes });
+					return response.accessToken;
+				}
```

maybe log if we get and unexpected value?

---

Changed this

---

sorry, I don't think I was explicit enough: what if the error is not InteractionRequiredAuthError or the type is not Popup/Redirect? I think we should log/throw for those totally unexpected cases

---

Thanks for pointing out. Added an else statement to throw the error.